File: app/src/main/java/com/asforce/asforcetkf2/util/SimpleTextWatcher.java
package com.asforce.asforcetkf2.util;

import android.text.Editable;
import android.text.TextWatcher;

/**
 * TextWatcher arayüzü için yardımcı sınıf
 * Sadece ihtiyaç duyulan metodu override etmek için kullanılır
 */
public abstract class SimpleTextWatcher implements TextWatcher {

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
        // Boş implementasyon
    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {
        // Boş implementasyon
    }

    @Override
    public abstract void afterTextChanged(Editable s);
}

File: app/src/main/java/com/asforce/asforcetkf2/util/SimpleTextWatcher.kt
package com.asforce.asforcetkf2.util

import android.text.Editable
import android.text.TextWatcher

/**
 * Simple implementation of TextWatcher that allows only overriding methods that are needed
 */
abstract class SimpleTextWatcher : TextWatcher {
    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
        // Default empty implementation
    }

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
        // Default empty implementation
    }

    override fun afterTextChanged(s: Editable) {
        // Default empty implementation
    }
} 